api/src/shared/application/scripts/base-script.js

```diff
@@ -0,0 +1,61 @@
+import 'dotenv/config';
+
+import Joi from 'joi';
+
+import { logger as defaultLogger } from '../../infrastructure/utils/logger.js';
+
+const META_INFO_SCHEMA = Joi.object({
+  description: Joi.string().required(),
+  permanent: Joi.boolean().required(),
+  options: Joi.object().optional(),
+});
+
+/**
+ * Base class for scripts with meta information and execution handling.
+ */
+export class BaseScript {
```

Comment:
Je trouve que `BaseScript` est trop générique, ça ne fait pas penser que c'est LA classe de base du code de Pix pour les scripts à exécuter pour faire des tâches particulières, en effet il y a des scripts pour faire tout et n'importe quoi … Je propose un autre nom `ManagedScript`, mais pourquoi pas `BaseManagedScript`  ou si vous avez d'autres idées : 

```suggestion
export class ManagedScript {
```

Reply:
> BaseScript est trop générique, ça ne fait pas penser que c'est LA classe de base

C'est littéralement ce que veut dire BaseScript non ?
`ManagedScript` ne me parle pas personnellement.
Je serai plus à l'aise avec une classe `Script` (si on ne garde pas `BaseScript`)

Reply:
J'ai également une préférence pour `Script` (au début, je l'avais appelé comme ça).